File: src/Services/Static/Documentation.cs
using DEA.Common.Extensions.DiscordExtensions;
using Discord.Commands;
using System.IO;
using System.Linq;

namespace DEA.Services.Static
{
    internal static class Documentation
    {
        public static void CreateAndSave(CommandService commandService)
        {
            var commandsDocumentation =
                "All commands are catagorized by modules. Each of the following sections is a module, and to gain more information about a " +
                "specific module, you may use the `$help [Module name]` command, or simply read below.\n\nThe syntax of the command usage is:\n\n" +
                "`Optional paramater: []`\n\n`Required paramater: <>`\n\n##Table Of Contents\n"; 

            var tableOfContents = string.Empty;
            var commandInfo = string.Empty;

            var sortedModules = commandService.Modules.OrderBy(x => x.Name);

            foreach (var module in sortedModules)
            {
                tableOfContents += $"- [{module.Name}](#{module.Name.ToLower()})\n";

                commandInfo += $"\n### {module.Name}\n";

                if (!string.IsNullOrWhiteSpace(module.Summary))
                {
                    commandInfo += $"\n{module.Summary}\n\n";
                }

                commandInfo += "Command | Description | Usage\n---------------- | --------------| -------\n";

                foreach (var command in module.Commands)
                {
                    commandInfo += $"{command.Name}|{command.Summary}|`${command.Name}{command.GetUsage()}`\n";
                }
            }

            commandsDocumentation += tableOfContents + commandInfo;

            File.WriteAllText(Config.MainDirectory + @"docs\docs\commands.md", commandsDocumentation);
        }
    }
}

File: src/services/Documentation.js
const fs = require('fs');
const path = require('path');
const assert = require('assert');
const config = require('../config.json');
const util = require('../utility');

class Documentation {
  async createAndSave(registry) {
    let commandsDocumentation = 'All commands are catagorized by groups. Each of the following sections is a group.\n\nThe syntax of the command usage is:\n\n`Optional paramater: []`\n\n`Required paramater: <>`\n\n##Table Of Contents\n'; 

    let tableOfContents = '';
    let commandInfo = '';

    const sortedGroups = registry.groups.sort(util.StringUtil.alphabeticallySort).values();

    for (const group of sortedGroups) {
      const formattedGroupName = util.StringUtil.upperFirstChar(group.name);
      
      tableOfContents += '- [' + formattedGroupName + '](#' + group.name.toLowerCase() + ')\n';

      commandInfo += '\n### '+ formattedGroupName +'\n';

      if (!util.StringUtil.isNullOrWhiteSpace(group.description)) {
        commandInfo += '\n' + group.description + '\n\n';
      }

      commandInfo += 'Command | Description | Usage\n---------------- | --------------| -------\n';

      for (const command of group.commands.sort(util.StringUtil.alphabeticallySort).values()) {
        commandInfo += util.StringUtil.upperFirstChar(command.name) + '|' + command.description + '|`' + config.prefix + command.getUsage() + '`\n';
      }
    }

    commandsDocumentation += tableOfContents + commandInfo;

    fs.writeFile(path.join(__dirname, '../../docs/docs/commands.md'), commandsDocumentation, (err) => {
      assert.equal(err, null);
    });
  }
}

module.exports = new Documentation();


